wire-ios/Wire-iOS/Sources/AppRootRouter.swift

```diff
@@ -392,16 +401,21 @@ extension AppRootRouter {
         Analytics.shared.provider?.selfUser = selfUser
     }
 
-    private func buildAuthenticatedRouter(account: Account, isComingFromRegistration: Bool) -> AuthenticatedRouter? {
-
+    private func buildAuthenticatedRouter(
+        account: Account,
+        userSession: UserSession,
+        isComingFromRegistration: Bool
+    ) -> AuthenticatedRouter? {
         let needToShowDataUsagePermissionDialog = appStateCalculator.wasUnauthenticated && !SelfUser.current.isTeamMember
 
-        return AuthenticatedRouter(rootViewController: rootViewController,
-                                   account: account,
-                                   selfUser: ZMUser.selfUser(),
-                                   isComingFromRegistration: isComingFromRegistration,
-                                   needToShowDataUsagePermissionDialog: needToShowDataUsagePermissionDialog,
-                                   featureRepositoryProvider: ZMUserSession.shared()!)
+        return AuthenticatedRouter(
+            rootViewController: rootViewController,
+            account: account,
+            userSession: userSession,
+            isComingFromRegistration: isComingFromRegistration,
+            needToShowDataUsagePermissionDialog: needToShowDataUsagePermissionDialog,
+            featureRepositoryProvider: ZMUserSession.shared()!
```

Comment:
can we do?
```suggestion
            featureRepositoryProvider: userSession
```

Reply:
We can't inject it here yet because `UserSession` isn't a featureRepositoryProvider...yet.